lib/everypolitician/popolo/collection.rb

```diff
@@ -32,7 +32,7 @@ def find_by(attributes = {})
       end
 
       def where(attributes = {})
-        attributes.map { |k, v| index_for(k.to_sym)[v] }.reduce(:&) || []
+          attributes.map { |k, v| index_for(k.to_sym)[v] }.reduce { |a, e| e.to_a & a } || []
```

Comment:
I fear that this line is getting a little too dense. Preventing `nil`s from ever reaching the `reduce` might be a _little_ cleaner (e.g. `attributes.map { |k, v| index_for(k.to_sym)[v] || [] }.reduce(:&)`), but I think it's still the wrong side of line-noise. 

I think there might be something useful to do around completely bailing out if we get no results for any of the searches, rather than even hitting the `reduce`, but I'm also a little concerned about going too far down _any_ particular version of this without knowing what's behind #70. Getting it working _right_ is probably the first step before making the code clean!